image_rotation.py

```diff
@@ -0,0 +1,172 @@
+'''
+Author: Kilando Chambers
+
+This module contains the image rotation class that can be used to rotate the
+an image containing a streak such that the streak is horizontal, or parllel with
+the x-axis.  Use the line_detection_testing.ipynb file to import this module and 
+apply it on multiple images.
+
+Process (in development currently):
+    1. Declare init variables (see description of parameters within the class)
+    2. Choose image
+    3. Process the image:
+        a. Apply z_score_trim once to reduce the outlier pixel values
+        b. Apply z_score_trim on this processed image again
+        c. Standardize and normalize the image using cv2
+        d. Perform Canny edge detection
+        e. Trim the edges of the image (if chosen)
+    4. Perform Hough Transformation on the Canny edge image to get streak coordinates
+    5. Convert streak polar coordinates into cartesian coordinates
+    6. Find the mean coordinates at the edges of the image
+    7. Determine the angle of rotation for the image based on the mean coordinates
+    8. Rotate the image at the determined angle of rotation
+
+Next steps from here:
+    Plot a histogram of the brightness of the image
+'''
+
+#Importing necessary libraries
+from tkinter import N
+import numpy as np
+import matplotlib.pyplot as plt
+import cv2
+
+#Create class with inheritance #
+class ImageRotation:
+    def __init__(self):
+        #considering if self.image and self.polar_coor should be required initialized variables
+        self.image = None
+        self.angle = None
+        self.coordinates = None
+        self.polar_coor = None
+        self.rotated_image = None
+        self.mean = None
+        self.median = None
+ 
+
+    def get_coord(self, rho, theta):
+        '''Finds Cartesian Coordinates of a line at the edge of image given the radius and angle
+        in Polar Coordinates
+        
+        Parameters
+        __________
+        rho : 'float'
+        Radius of detected line in Polar Coordinates
+        
+        theta : 'float'
+        Angle of detected line in Polar Coordinates
+        '''
+
+        x_size = self.image.shape[1]
+        y_size = self.image.shape[0]
+
+        #first guess coordinates if they are within the boundaries of the image
+        x1, y1 = (0, rho / np.sin(theta))
+        x2, y2 = (rho / np.cos(theta), 0)
+
+        slope = y1 / x2
+
+        #adjust coordinates to certainly fit within the boundaries of the image
+        if x2 > x_size:
+            x2 = x_size
+            y2 = (-1 * (np.cos(theta) / np.sin(theta)) * x2) + (rho / np.sin(theta))
+            #formula can be found on opencv2 Hough Transform tutorial
```

Comment:
A link would be better, best would be to jot down the formula when it's this simple because links can disappear too.